### PR TITLE
Fix BoxPlot styling issue

### PR DESCRIPTION
**Pull request by kalletlak**
1. Reduce height of zero count bars
2. Fix Axis offset
3. Reduce white space between groups
4. Reduce bar width if there are more than 10 categories

Screen Shot 2019-03-25 at 9 48 54 AM

### CODE REVIEW

File: src/shared/components/plots/MultipleCategoryBarPlot.tsx [hunk 1]
```diff
@@ -309,9 +309,14 @@ export default class MultipleCategoryBarPlot extends React.Component<IMultipleCa
             if (this.props.stacked) {
                 numBars = this.data[0].counts.length;
             } else {
-                numBars = _.sumBy(this.data, categoryPlotData => categoryPlotData.counts.length);
+                //majorCategories*minorCategories
+                numBars = this.data[0].counts.length * this.data.length;
+                //remove additional padding added by barSeparation
```
Comment (adamabeshouse) [inline src/shared/components/plots/MultipleCategoryBarPlot.tsx]: wait if this is a problem then we should just remove barSeparation. I'm confused why would we add and then remove space?

File: src/shared/components/plots/MultipleCategoryBarPlot.tsx [hunk 2]
```diff
@@ -330,7 +335,9 @@ export default class MultipleCategoryBarPlot extends React.Component<IMultipleCa
     }
 
     @computed get barWidth() {
-        return this.props.barWidth || 10;
+        let barWidth = this.props.barWidth || 10;
+        //if number of minorCategories greater than 10 then reduce the width by half
+        return this.props.stacked ? barWidth : this.data.length > 10 ? barWidth/2 : barWidth;
```
Comment (adamabeshouse) [inline src/shared/components/plots/MultipleCategoryBarPlot.tsx]: can you write it out? I think nested ternarys are harder to read

File: src/shared/components/plots/MultipleCategoryBarPlot.tsx [hunk 3]
```diff
@@ -355,8 +362,18 @@ export default class MultipleCategoryBarPlot extends React.Component<IMultipleCa
...
+                if (c.count === 0) {
+                    addOffset = true;
+                    break;
+                }
+            }
+            if (addOffset) break;
+        }
         //add a small offset when its not a stacked bar to show empty bar
-        return this.props.stacked ? 0 : (0.03 * (this.maxMajorCount / NUM_AXIS_TICKS));
+        return this.props.stacked ? 0 : addOffset ? (0.01 * (this.maxMajorCount / NUM_AXIS_TICKS)) : 0;
```
Comment (adamabeshouse) [inline src/shared/components/plots/MultipleCategoryBarPlot.tsx]: expand nested ternary please

**Comment by adamabeshouse** [comment]
there should be zero bars here, right?

image

**Comment by kalletlak** [comment]
there is, I reduced its height

**Comment by adamabeshouse** [comment]
I dont see any zero bars?

**Comment by kalletlak** [comment]
ok, I'll revert that back

File: src/shared/components/plots/MultipleCategoryBarPlot.tsx [hunk 4]
```diff
@@ -326,11 +330,17 @@ export default class MultipleCategoryBarPlot extends React.Component<IMultipleCa
     }
 
     @computed get barSeparation() {
-        return 0.2*this.barWidth;
+        return this.props.stacked ? 0.2 * this.barWidth : 0;
```
Comment (adamabeshouse) [inline src/shared/components/plots/MultipleCategoryBarPlot.tsx]: :+1:

**Pull request merged**